easybuild/easyblocks/f/ferret.py

```diff
@@ -48,80 +53,156 @@ def configure_step(self):
+                "external_functions/ef_utility/site_specific.mk",
+            ]
+        else:
+            fns = "site_specific.mk"
 
-        for line in fileinput.input(fn, inplace=1, backup='.orig'):
+        for fn in fns:
+            for line in fileinput.input(fn, inplace=1, backup='.orig'):
+                line = re.sub(r"^BUILDTYPE\s*=.*", "BUILDTYPE = %s" % buildtype, line)
+                line = re.sub(r"^INSTALL_FER_DIR =.*", "INSTALL_FER_DIR = %s" % self.installdir, line)
 
-            line = re.sub(r"^BUILDTYPE\s*=.*", "BUILDTYPE = %s" % buildtype, line)
-            line = re.sub(r"^INSTALL_FER_DIR =.*", "INSTALL_FER_DIR = %s" % self.installdir, line)
+                for name in deps:
+                    line = re.sub(r"^(%s.*DIR\s*)=.*" % name.upper(), r"\1 = %s" % get_software_root(name), line)
 
-            for name in deps:
-                line = re.sub(r"^(%s.*DIR\s*)=.*" % name.upper(), r"\1 = %s" % get_software_root(name), line)
+                if LooseVersion(self.version) >= LooseVersion("7.3"):
+                    line = re.sub(r"^DIR_PREFIX =.*", "DIR_PREFIX = %s" % self.cfg['start_dir'], line)
+                    line = re.sub(r"^FER_LOCAL_EXTFCNS = $(FER_DIR)",
+                                  "FER_LOCAL_EXTFCNS = $(INSTALL_FER_DIR)/libs", line)
 
-            sys.stdout.write(line)
+                sys.stdout.write(line)
 
         comp_vars = {
-            'CC':'CC',
-            'CFLAGS':'CFLAGS',
-            'CPPFLAGS':'CPPFLAGS',
-            'FC':'F77',
+            'CC': 'CC',
+            'CFLAGS': 'CFLAGS',
+            'CPPFLAGS': 'CPPFLAGS',
+            'FC': 'F77',
+        }
+
+        gfort2ifort = {
+            '-fno-second-underscore': ' ',
+            '-fno-backslash': ' ',
+            '-fdollar-ok': ' ',
+            '-ffast-math': ' ',
+            '-ffixed-line-length-132': '-132',
+            '-fno-automatic': ' ',
+            '-ffpe-trap=overflow': ' ',
+            '-fimplicit-none': '-implicitnone',
+            '-fdefault-real-8': '-r8',
+            '-fdefault-double-8': ' ',
+            '-Wl,-Bstatic -lgfortran -Wl,-Bdynamic': ' ',
+            '-v --verbose -m64': ' ',
+            '-export-dynamic': ' ',
+            '-DG77_SIGNAL': ' ',
         }
 
         fn = 'xgks/CUSTOMIZE.%s' % buildtype
 
         for line in fileinput.input(fn, inplace=1, backup='.orig'):
 
-            for x,y in comp_vars.items():
-                line = re.sub(r"^(%s\s*)=.*" % x, r"\1=%s" % os.getenv(y), line)
+            for x, y in comp_vars.items():
+                line = re.sub(r"^(%s\s*)=.*" % x, r"\1='%s'" % os.getenv(y), line)
 
             line = re.sub(r"^(FFLAGS\s*=').*-m64 (.*)", r"\1%s \2" % os.getenv('FFLAGS'), line)
-            line = re.sub(r"^(LD_X11\s*)=.*", r"\1='-L/usr/lib64/X11 -lX11'", line)
+            if LooseVersion(self.version) >= LooseVersion("7.3"):
+                line = re.sub(r"^(LD_X11\s*)=.*", r"\1='-L$(EBROOTX11)/lib -lX11'", line)
```

Comment:
Why not inject the actual value from `$EBROOTX11` (obtained via `get_software_root('X11')`?